sycl/CMakeLists.txt

```
@@ -34,12 +34,20 @@ set(CLANG_VERSION "${CLANG_VERSION_MAJOR}.${CLANG_VERSION_MINOR}.${CLANG_VERSION
 set(LLVM_INST_INC_DIRECTORY "lib${LLVM_LIBDIR_SUFFIX}/clang/${CLANG_VERSION}/include")
 set(dst_dir ${LLVM_LIBRARY_OUTPUT_INTDIR}/clang/${CLANG_VERSION}/include)
 
-find_package(OpenCL)
+# Find OpenCL headers and libraries installed in the system and use them to
+# build SYCL runtime.
+# WARNING: use with caution, building SYCL runtime with OpenCL implementation
+# instead of Khronos ICD loader might cause build and/or portability issues.
+option(BUILD_WITH_SYSTEM_OPENCL OFF)
+
+if( (OpenCL_INCLUDE_DIR AND OpenCL_LIBRARY) OR BUILD_WITH_SYSTEM_OPENCL)
```

```suggestion
if(BUILD_WITH_SYSTEM_OPENCL)
```
I think we don't need to call `find_package` if these variables are set. If I understand correctly, their values might be reset(!) by this function.
The only value of `find_package` in this case is correctly set `OpenCL::OpenCL` target.
@valerydmit added similar target, but I think it requires a couple of adjustments.

Please, make sure that in our cmake OpenCL target is configured like this: https://github.com/Kitware/CMake/blob/release/Modules/FindOpenCL.cmake#L160